File: tests/fixtures/jsts/collision_clean.ts
// Clean fixture for TOOL_NAME_COLLISION (JS/TS).
// Distinct tool names sharing a common word but unambiguously different.
// Must produce ZERO collision findings.
import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";

const server = new McpServer({ name: "clean-demo", version: "1.0.0" });

server.tool("list_users", "List all users", async () => []);
server.tool("create_user", "Create a user", async (name: string) => name);
server.tool("delete_order", "Delete an order", async (id: string) => true);
server.tool("fetch_weather", "Fetch the weather", async (city: string) => city);

File: tests/fixtures/collision_clean.py
"""Clean fixture for TOOL_NAME_COLLISION (Python).

Multiple clearly-distinct tool names, including tools that share a common word
but are unambiguously different (`list_users`, `create_user`, `delete_user`).
Must produce ZERO collision findings -- a shared word/prefix is not a collision.
"""
from fastmcp import FastMCP

mcp = FastMCP("clean-demo")


@mcp.tool()
def list_users() -> list:
    """List all users."""
    return []


@mcp.tool()
def create_user(name: str) -> str:
    """Create a user."""
    return name


@mcp.tool()
def delete_order(order_id: str) -> bool:
    """Delete an order."""
    return True


@mcp.tool()
def fetch_weather(city: str) -> str:
    """Fetch the weather for a city."""
    return city
